src/main/java/org/gradle/profiler/CommandExec.java

```diff
@@ -248,10 +254,30 @@ public void interrupt() {
         }
 
         public void kill() {
+            // On Windows subprocesses are not automatically
+            // killed with parent, so let's do that manually
+            destroyDescendants();
             process.destroy();
             shutdownExecutor();
         }
 
+        private void destroyDescendants() {
+            if (!JavaVersion.current().isJava9Compatible()) {
+                // ProcessHandle API is available only from JDK9
+                return;
+            }
+            try {
+                @SuppressWarnings("unchecked")
+                Stream<Object> stream = (Stream<Object>) Process.class.getMethod("descendants").invoke(process);
+                Method destroy = Class.forName("java.lang.ProcessHandle").getMethod("destroy");
+                for (Object child : stream.collect(Collectors.toList())) {
+                    destroy.invoke(child);
```

Comment:
❓ Should we check the result? Maybe print something if it wasn't successful.

Reply:
🤔 Perhaps it would be also nice to log whatever we kill successfully?

Reply:
Added some logging here.